doc/designs/disable-stale-users.md

```diff
@@ -0,0 +1,287 @@
+
+If the requirement is to disable users not seen for 90 days maximum, updating
+the attribute every 35 days is good enough.
+At worst, the user would have authenticated the first day and at least once
+during this 35 day period, say day #29. The attribute would therefore contain
+the date from the first day, as day #29 would have been considered too close to
+the existing timestamp to warrant updating it. The user account would then be
+disabled after 61 days of inactivity which fits (loosely) the requirement.
+In short, having a 90-day inactive user disabling requirement would by default
+disable accounts inactive for 55 (90-35) to 90 days.
+
+
+## Limitations
+
+### Environments with mandatory, periodic password changes
+
+If regulations call for stale/inactive accounts to be disabled within X
+number of days, then the lifetime of the passwords for these accounts must be
+set to a value lower than:
+* X,
+* minus the expected grace period mentioned above,
+* minus the maximum amount of time between two invocations of the DSU tool,
+* minus the expected maximum replication delay from the replica the DSU tool is
+installed on to the farthest point of the cluster.
+
+### Environments with no mandatory password changes
+
+The longer the time period, the less update activity for the timestamp
+attribute. But using a longer period runs the risk of disabling users who have
+been gone for too short a time. For instance, setting the update period to 65
+days would disable accounts inactive for the last 25 to 90 days. Whether this
+fits the site's requirements is up to the administrators to determine.
+
+Sites requiring a better accuracy must take care not to set the update period
+too low for performance reasons. It is strongly recommended to make sure the
+update period is longer than the expected maximum PTO or simply "away from IPA"
+period for most user accounts. This is especially true when these away periods
+are in sync, like in schools.
+
+
+## Multiple runners
+
+DSU can be installed on more than a single replica. This provides redundancy.
+Having multiple DSU instances active at the same time provides no advantage,
+so the systemd timer helper script includes a randomized time wait period.
+
+On very busy clusters, using dedicated replicas (hidden replicas) for DSU is
+recommended.
+
+
+## How to Use DSU
+
+### Installation
+
+The Disable Stale Users tooling is contained in the freeipa-server-dsu
+package. 
+``dnf install freeipa-server-dsu`` will install that package.
+
+
+### Configuration
+
+#### Discussion
+
+There are two ways to store configuration in FreeIPA, the first way via
+LDAP attributes, the second way via configuration files. The proposal below
+uses configuration files but this is not set in stone yet.
+
+#### Environments with mandatory, periodic password changes
+
+The configuration file for DSU contains the following variables:
+
+```
+[password-based-dsu]
+disable_accounts = False
+grace_period = 7d
```

Comment:
disable_accounts and grace_period were moved to LDAP? Or just grace_period?

Reply:
Both of them now, thanks!